models/port.js

```diff
@@ -80,33 +80,102 @@ module.exports = (sequelize, DataTypes) => {
                 }
             },
             private_key: {
-                type: DataTypes.BLOB,
+                type: DataTypes.TEXT,
                 set(val) {
-                    this.setDataValue('private_key', val ? crypto.encrypt(val.toString()) : val);
+                    if (val) {
+                        try {
+                            val = crypto.encrypt(val)
+                        }
+                        catch (e) {
+                            val = ""
+                        }
+                    }
+                    else {
+                        val = ""
+                    }
+                    this.setDataValue('private_key', val);
                 },
                 get() {
                     let val = this.getDataValue('private_key');
-                    return val ? crypto.decrypt(val) : "";
+                    if (val) {
```

Comment:
So I guess here is the logic that we need to add to decrypt and encrypt. If the value is falsy, currently we return the value, but instead we should return an empty string. Then we could remove all this extra stuff.